modules/assim.sequential/R/sda.enkf.R

```diff
@@ -3,49 +3,56 @@
 ##' @author Michael Dietze and Ann Raiho \email{dietze@@bu.edu}
 ##' 
 ##' @param settings    PEcAn settings object
-##' @param IC          data.frame of initial condition sample (nens X nstate)
-##' @param prior       data.frame of model parameter sample (nens X nstate)
 ##' @param obs.mean    data.frame of observations of the mean of variables (time X nstate)
 ##' @param obs.cov      data.frame of observations of the sd of variables (time X nstate)
-##' @param processvar  flag for if process variance should be estimated or not
-##' @param sample.parameters flag for if parameters should come from meta.analysis or be defaults. if false parameters are set to defaults
+##' @param pick.trait.params  character vector of traits to sample if sample.parameters == TRUE
+##' @param given.process.variance flag for if parameters should come from meta.analysis or be defaults. if false parameters are set to defaults
 ##' 
 ##' @description State Variable Data Assimilation: Ensemble Kalman Filter
 ##' 
 ##' @return NONE
 ##' 
-sda.enkf <- function(settings, IC, prior, obs.mean, obs.cov, variables,
-                     processvar=FALSE, sample_parameters=FALSE,
-                     pick.trait.params, given.process.variance){
+sda.enkf <- function(settings, obs.mean, obs.cov, pick.trait.params = NULL,
+                     given.process.variance = NULL,IC = NULL){
+  
+  ###-------------------------------------------------------------------###
+  ### read settings                                                     ###
+  ###-------------------------------------------------------------------### 
   
-  ## settings
   model <- settings$model$type
   write <- settings$database$bety$write
   defaults <- settings$pfts
-  outdir <- settings$host$outdir #settings$run$host$outdir
-  rundir <- settings$host$rundir #run$
+  outdir <- settings$host$outdir
+  rundir <- settings$host$rundir
   host <- settings$host
-  forecast.duration <- 1 #eventually in settings #try to run with dates
-  forecast.time.step <- 1 #eventually in settings #dt
-  spin.up <- 90 #eventually in settings
-  nens = settings$ensemble$size#30#nrow(IC) #right?
-  start.year <- strftime(settings$run$start.date,"%Y")
-  end.year   <- strftime(settings$run$end.date,"%Y")
-  processvar #need to be in xml
-  sample_parameters #need to be xml
-  
-  ###HACK ###LINKAGES SPECIFIC
-  new.met <- paste0(rundir,"/climate.Rdata")
-  
-  #   if(nrow(prior) == 1 | is.null(nrow(prior))){
-  #     var.names = names(prior)
-  #     prior = as.data.frame(matrix(rep(prior,each=nens),nrow=nens))
-  #     names(prior) = var.names
-  #   }
+  forecast.time.step <- settings$state.data.assimilation$forecast.time.step
+  spin.up.start <- strftime(settings$state.data.assimilation$spin.up$start.date,"%Y")
+  spin.up.end <- strftime(settings$state.data.assimilation$spin.up$end.date,"%Y")
+  nens = settings$ensemble$size#settings$state.data.assimilation$n.ensemble #which one?
```

Comment:
It's be better to get the ensemble size from settings$state.data.assimilation$n.ensemble because there's nothing that says the ensemble analysis and SDA should have the same ensemble size, plus specifying ensemble$size has the unwanted consequence of turning on the EA in the workflow (not what we want)